lib/IDE/CodeCompletion.cpp

```diff
@@ -4170,50 +4170,78 @@ class CompletionOverrideLookup : public swift::VisibleDeclConsumer {
       return nullptr;
 
     Type ResultT;
-    if (auto *FD = dyn_cast<FuncDecl>(VD))
+    if (auto *FD = dyn_cast<FuncDecl>(VD)) {
+      if (FD->getGenericParams()) {
+        // Generic function cannot have opaque result type.
```

Comment:
Is this really the case? This seems to type check and compile just fine:

```
func foo<T>(_ t: T) -> some Any {
  return t
}
```

Reply:
This code is only for conformance completion.

```swift
protocol MyProtocol {}
class C: MyProtocol {}

protocol P {
  associatedtype Assoc : MyProtocol
  func foo<T>(x: T) -> Assoc
}
struct S: P {
  func foo<T>(x: T) -> some MyProtocol { return C() }
}
```
This `S.foo(x:)` cannot be used to infer `S.Assoc`.